standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -1176,6 +1186,55 @@ protected void updateWSCommitIdAndCleanUpMetadata(Statement stmt, long txnid, Tx
               TxnDbUtil.getEpochFn(dbProduct) + " WHERE \"CQ_TXN_ID\" = " + txnid);
     }
   }
+
+  private CompactionInfo getCompactionByTxnId(Connection dbConn, long txnid) throws SQLException, MetaException {
+    CompactionInfo info = null;
```

Comment:
Wouldn't it be better if we wrap this with Optional?

Reply:
good catch, fixed.